### PR TITLE
♻️  Cleanup in preparation for new payment plugin

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

In order to simplify the review of PR https://github.com/ITISFoundation/osparc-simcore/pull/4683, this PR addresses some minor cleanup on 
- wallets plugin 
- uses `arrow` for `utcnow`
- moves API schemas to `models_library`
- fixes `HEAD`-related entry in `openapi.yaml`
- common fixtures for `wallets`
- new basic type for string identifiers that checks for empty strings: `IDStr`
- Ruff-ied some files

## Related issue/s

- preparation for #4657

## How to test
 in place

## DevOps
None

### CODE REVIEW

File: packages/models-library/src/models_library/rabbitmq_messages.py [hunk 1]
```diff
@@ -181,7 +182,7 @@ class RabbitResourceTrackingBaseMessage(RabbitMessageBase):
         ..., description="uniquely identitifies the service run"
     )
     created_at: datetime.datetime = Field(
-        default_factory=lambda: datetime.datetime.now(datetime.timezone.utc),
+        default_factory=lambda: arrow.utcnow(),
```
Comment (sanderegg) [inline packages/models-library/src/models_library/rabbitmq_messages.py:185]: hmm, I do not think that this returns a datetime object.
don't you need to call ```arrow.utcnow().datetime``` ????

File: packages/models-library/src/models_library/wallets.py [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 
+from .utils.enums import StrAutoEnum
+
 WalletID: TypeAlias = PositiveInt
 
 
-class WalletStatus(str, enum.Enum):
-    ACTIVE = "ACTIVE"
-    INACTIVE = "INACTIVE"
+class WalletStatus(StrAutoEnum):
```
Comment (sanderegg) [inline packages/models-library/src/models_library/wallets.py:12]: thx

**Review by sanderegg — APPROVED** [review]
please check your change in rabbitmq_messages before pushing this. Thanks!

File: packages/models-library/src/models_library/api_schemas_webserver/wallets.py [hunk 3]
```diff
@@ -25,3 +24,16 @@ class WalletGetPermissions(WalletGet):
     read: bool
     write: bool
     delete: bool
+
+
+class CreateWalletBodyParams(OutputSchema):
+    name: str
+    description: str | None = None
+    thumbnail: str | None = None
```
Comment (matusdrobuliak66) [inline packages/models-library/src/models_library/api_schemas_webserver/wallets.py:32]: thanks!

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

File: packages/models-library/src/models_library/rabbitmq_messages.py (hunk 1 above)
Comment (pcrespov) [reply packages/models-library/src/models_library/rabbitmq_messages.py:185]: good catch! thx

File: services/web/server/src/simcore_service_webserver/api/v0/openapi.yaml [hunk 4]
```diff
@@ -8656,17 +8670,17 @@ components:
...
       - created
       - modified
-      - available_credits
+      - availableCredits
       type: object
       properties:
-        wallet_id:
-          title: Wallet Id
+        walletId:
+          title: Walletid
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/api/v0/openapi.yaml:8683]: yes. This is a convention we are suppose to fulfill for the web-sever api

Aha ... we should fix it in the front-end probably! let me check!

[the remark this answers is not in the document]
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/api/v0/openapi.yaml:8683]: yes. @odeimaiz  I applied the changes in the openapi specs to the front-end

**Review by odeimaiz — APPROVED** [review]
The js part looks good, thanks!

File: packages/models-library/src/models_library/basic_types.py [hunk 5]
```diff
@@ -44,6 +44,12 @@ class UUIDStr(ConstrainedStr):
     regex = re.compile(UUID_RE)
 
 
+# non-empty string identifier e.g. "123" or "name_id1" (avoids "" identifiers)
+class IDStr(ConstrainedStr):
```
Comment (GitHK) [inline packages/models-library/src/models_library/basic_types.py:48]: You could summarise the comment inside the name of the class? Personally I would not mind a longer class name here, like: `NonEmptyIDString`

**Review by GitHK — APPROVED** [review]

**Pull request merged**